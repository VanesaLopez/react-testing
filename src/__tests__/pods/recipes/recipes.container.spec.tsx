import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Recipe } from 'pods/recipes/recipes.view-model';
import { RecipesPodState, recipesPodReducer } from 'pods/recipes/store/reducers';
import { RecipesContainer } from 'pods/recipes/recipes.container';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { State } from 'core/store/root-reducer';
import * as actions from 'pods/recipes/store';


describe('Tests Recipes Container', () => {
  it('should render as expected', () => {
    // Arrange
    const recipes: Recipe[] = [
      { id: 'xx', name: 'pizza', ingredients: ['tomato', 'cheese', 'flour']},
      { id: 'yy', name: 'cake', ingredients: ['cream', 'butter', 'egg', 'flour']}
    ];

    const initalState: Partial<State> = {
      recipesPod: {
        recipes,
      },
    };

    // Act
    const { asFragment } = renderWithRedux(<RecipesContainer />, {
      initalState,
      reducer: recipesPodReducer,
    });

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

  it('Table should be 0 elements', () => {
    // Arrange
    const recipes: Recipe[] = [];

    const initalState: Partial<State> = {
      recipesPod: {
        recipes,
      },
    };

    // Act
    const { queryAllByTestId } = renderWithRedux(<RecipesContainer />, {
      initalState,
      reducer: recipesPodReducer,
    });
    const elements = queryAllByTestId('row');

    // Assert
    expect(elements).toHaveLength(0);
  });

  it('Table should be 2 elements', () => {
    // Arrange
    const recipes: Recipe[] = [
      { id: 'xx', name: 'pizza', ingredients: ['tomato', 'cheese', 'flour']},
      { id: 'yy', name: 'cake', ingredients: ['cream', 'butter', 'egg', 'flour']}
    ];

    const initalState: Partial<State> = {
      recipesPod: {
        recipes,
      },
    };

    // Act
    const { queryAllByTestId } = renderWithRedux(<RecipesContainer />, {
      initalState,
      reducer: recipesPodReducer,
    });
    const elements = queryAllByTestId('row');

    // Assert
    expect(elements).toHaveLength(2);
  });

  it('Table should be 2 elements', () => {
    // Arrange
    const recipes: Recipe[] = [
      { id: 'xx', name: 'pizza', ingredients: ['tomato', 'cheese', 'flour']},
      { id: 'yy', name: 'cake', ingredients: ['cream', 'butter', 'egg', 'flour']}
    ];
    const initalState: Partial<State> = {
      recipesPod: {
        recipes,
      },
    };
    const getRecipesRequestStub = jest.spyOn(actions, 'getRecipesRequest');

    // Act
    renderWithRedux(<RecipesContainer />, {
      initalState,
      reducer: recipesPodReducer,
    });

    // Assert
    expect(getRecipesRequestStub).toHaveBeenCalled();
  });
});

const renderWithRedux = (
  component,
  { initalState = {} , reducer, store = createStore(reducer, initalState) }) => (
    {
      ...render(<Provider store={store}>{component}</Provider>),
      store
    }
);