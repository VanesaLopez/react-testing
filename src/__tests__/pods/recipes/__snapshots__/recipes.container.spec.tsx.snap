// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Recipes Container should render as expected 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-container-1"
  >
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-textField-2 MuiFormControl-marginNormal"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
      >
        Search
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-85 MuiPaper-rounded"
    >
      <table
        class="MuiTable-root makeStyles-table-86"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              Ingredients
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root makeStyles-row-87"
            data-testid="row"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              scope="row"
            >
              pizza
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              tomato , cheese , flour
            </td>
          </tr>
          <tr
            class="MuiTableRow-root makeStyles-row-87"
            data-testid="row"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              scope="row"
            >
              cake
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              cream , butter , egg , flour
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
